src/simulation/ApplyLoad.cpp

```diff
@@ -0,0 +1,240 @@
+    ledgerUpgrade.newConfig() = upgradeSetKey;
+    auto v = xdr::xdr_to_opaque(ledgerUpgrade);
+    upgrade.push_back(UpgradeType{v.begin(), v.end()});
+
+    closeLedger({invokeTx.second}, upgrade);
+}
+
+void
+ApplyLoad::setupLoadContracts()
+{
+    auto wasm = rust_bridge::get_test_wasm_loadgen();
+    xdr::opaque_vec<> wasmBytes;
+    wasmBytes.assign(wasm.data.begin(), wasm.data.end());
+
+    LedgerKey contractCodeLedgerKey;
+    contractCodeLedgerKey.type(CONTRACT_CODE);
+    contractCodeLedgerKey.contractCode().hash = sha256(wasmBytes);
+
+    mLoadCodeKey = contractCodeLedgerKey;
+
+    auto const& lm = mApp.getLedgerManager();
+    auto uploadTx = createUploadWasmTransaction(
+        lm.getLastClosedLedgerNum() + 1, 0, wasmBytes, contractCodeLedgerKey,
+        std::nullopt);
+
+    closeLedger({uploadTx.second});
+
+    for (auto kvp : mAccounts)
+    {
+        auto salt = sha256("Load contract " + std::to_string(kvp.first));
+
+        auto createTx = createContractTransaction(
+            lm.getLastClosedLedgerNum() + 1, 0, contractCodeLedgerKey,
+            wasmBytes.size() + 160, salt, std::nullopt);
+        closeLedger({createTx.second});
+
+        auto instanceKey =
+            createTx.second->sorobanResources().footprint.readWrite.back();
+
+        ContractInstance instance;
+        instance.readOnlyKeys.emplace_back(mLoadCodeKey);
+        instance.readOnlyKeys.emplace_back(instanceKey);
+        instance.contractID = instanceKey.contractData().contract;
+        mLoadInstances.emplace(kvp.first, instance);
+    }
+}
+
+void
+ApplyLoad::benchmark()
+{
+    auto& lm = mApp.getLedgerManager();
+    std::vector<TransactionFrameBasePtr> txs;
+
+    auto resources =
+        multiplyByDouble(lm.maxLedgerResources(true),
+                         2 /*TODO: use TRANSACTION_QUEUE_SIZE_MULTIPLIER*/);
+
+    std::vector<uint64_t> shuffledAccounts(mAccounts.size());
+    std::iota(shuffledAccounts.begin(), shuffledAccounts.end(), 0);
+    stellar::shuffle(std::begin(shuffledAccounts), std::end(shuffledAccounts),
+                     gRandomEngine);
+
+    for (auto accountIndex : shuffledAccounts)
+    {
+        auto it = mAccounts.find(accountIndex);
+        releaseAssert(it != mAccounts.end());
+
+        auto instanceIter = mLoadInstances.find(it->first);
+        releaseAssert(instanceIter != mLoadInstances.end());
+        auto const& instance = instanceIter->second;
+        auto tx = invokeSorobanLoadTransaction(
+            lm.getLastClosedLedgerNum() + 1, it->first, instance,
+            rust_bridge::get_write_bytes().data.size() + 160, std::nullopt);
+
+        if (!anyGreater(tx.second->getResources(false), resources))
+        {
+            resources -= tx.second->getResources(false);
+        }
+        else
+        {
+            break;
+        }
+
+        txs.emplace_back(tx.second);
+    }
+
+    closeLedger(txs);
+
+    CLOG_INFO(Perf, "mApplySorobanSuccess count {}",
```

Comment:
What's the success rate currently? I guess it would be convenient to output the rate itself as well, also maybe as an explicit return value of the function. Then the caller may assert on the success rate being at least X without needing to read the logs.

Reply:
Updated to include a method that returns the overall success rate. The success rate is above 99.5%.